[EuiSuperDatePicker] Add `customQuickSelectRender` prop (#6382)
elastic/eui: upcoming_changelogs/6382.md

> - Added the `customQuickSelectRender` prop to `EuiSuperDatePicker` which allows the Quick Select panels to be rendered in a custom order 

Reviewer: copy feedback - I tweaked it slightly because this new functionality allows for more than just re-ordering
```suggestion
- Added the `customQuickSelectRender` render prop to `EuiSuperDatePicker`, which allows customizing the Quick Select popover
```
